go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -756,7 +756,7 @@ func (call *builtinHour) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Int64, Col: collationNumeric, Flag: arg.Flag | flagNullable}, nil
 }
 
-func yearDayToTime(y, yd int64) time.Time {
+func yearDayToTime(env *ExpressionEnv, y, yd int64) time.Time {
```

Comment:
I think it’s best to pass in the most specific type. So we should pass in the time zone itself here I think. 

Same applies to lastday as well.

Reply:
done.